File: scripts/clean_data.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import json
import os
import sys

def clean_data():
    """清理數據，統一描述格式"""
    try:
        print('🧹 開始清理數據...')
        
        # 讀取原始數據
        data_path = '../data/data.json'
        with open(data_path, 'r', encoding='utf-8') as f:
            data = json.load(f)
        
        print(f'📊 原始記錄數: {len(data["records"])}')
        
        cleaned_count = 0
        skipped_count = 0
        
        # 清理每筆記錄
        for record in data['records']:
            if not record.get('description'):
                skipped_count += 1
                continue
            
            # 檢查是否需要清理
            needs_cleaning = ('-' not in record['description'] and 
                            record.get('mainCategory'))
            
            if needs_cleaning:
                # 將格式從 "描述" 改為 "主類別-描述"
                old_description = record['description']
                record['description'] = f"{record['mainCategory']}-{old_description}"
                cleaned_count += 1
                
                print(f'🔄 清理記錄 {record["id"]}: "{old_description}" → "{record["description"]}"')
            else:
                skipped_count += 1
        
        # 保存清理後的數據
        with open(data_path, 'w', encoding='utf-8') as f:
            json.dump(data, f, ensure_ascii=False, indent=2)
        
        print('✅ 數據清理完成！')
        print(f'📈 統計結果:')
        print(f'   - 總記錄數: {len(data["records"])}')
        print(f'   - 已清理: {cleaned_count} 筆')
        print(f'   - 跳過: {skipped_count} 筆')
        print(f'   - 備份文件: data_backup.json')
        
    except Exception as error:
        print(f'❌ 數據清理失敗: {error}')
        sys.exit(1)

if __name__ == '__main__':
    clean_data()

File: scripts/clean_data.js
#!/usr/bin/env node

// 數據清理腳本 - 統一描述格式
const fs = require('fs').promises;
const path = require('path');

async function cleanData() {
    try {
        console.log('🧹 開始清理數據...');
        
        // 讀取原始數據
        const dataPath = path.join(__dirname, '../data/data.json');
        const dataContent = await fs.readFile(dataPath, 'utf8');
        const data = JSON.parse(dataContent);
        
        console.log(`📊 原始記錄數: ${data.records.length}`);
        
        // 備份原始數據
        const backupPath = path.join(__dirname, '../data/data_backup.json');
        await fs.writeFile(backupPath, dataContent, 'utf8');
        console.log('💾 已備份原始數據到 data_backup.json');
        
        let cleanedCount = 0;
        let skippedCount = 0;
        
        // 清理每筆記錄
        for (const record of data.records) {
            if (!record.description) {
                skippedCount++;
                continue;
            }
            
            // 檢查是否需要清理
            const needsCleaning = !record.description.includes('-') && record.mainCategory;
            
            if (needsCleaning) {
                // 將格式從 "描述" 改為 "主類別-描述"
                const oldDescription = record.description;
                record.description = `${record.mainCategory}-${oldDescription}`;
                cleanedCount++;
                
                console.log(`🔄 清理記錄 ${record.id}: "${oldDescription}" → "${record.description}"`);
            } else {
                skippedCount++;
            }
        }
        
        // 保存清理後的數據
        await fs.writeFile(dataPath, JSON.stringify(data, null, 2), 'utf8');
        
        console.log('✅ 數據清理完成！');
        console.log(`📈 統計結果:`);
        console.log(`   - 總記錄數: ${data.records.length}`);
        console.log(`   - 已清理: ${cleanedCount} 筆`);
        console.log(`   - 跳過: ${skippedCount} 筆`);
        console.log(`   - 備份文件: data_backup.json`);
        
    } catch (error) {
        console.error('❌ 數據清理失敗:', error);
        process.exit(1);
    }
}

// 執行清理
cleanData();
